Can not build shared library (dll) on MSVC with cmake

I very much like the cmake based build instructions, they work quite well on many platforms. But recently I tried to switch from a static to a shared library build. Since then I receive the following error on MSVC with Visual Studio 2017 and 2019:
```
[...]
 [10/10] cmd.exe /C "cd . && C:\data\Tools\bin\cmake.exe -E vs_link_exe --intdir=CMakeFiles\lz4cli.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100183~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100183~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1426~1.288\bin\Hostx64\x64\link.exe  CMakeFiles\lz4cli.dir\C_\lz4\programs\bench.c.obj CMakeFiles\lz4cli.dir\C_\lz4\programs\lz4cli.c.obj CMakeFiles\lz4cli.dir\C_\lz4\programs\lz4io.c.obj CMakeFiles\lz4cli.dir\C_\lz4\programs\datagen.c.obj  /out:lz4.exe /implib:lz4.lib /pdb:lz4.pdb /version:0.0  /machine:x64 /MACHINE:X64 /DEBUG /debug /INCREMENTAL /subsystem:console  lz4.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
 FAILED: lz4.exe 
 cmd.exe /C "cd . && C:\data\Tools\bin\cmake.exe -E vs_link_exe --intdir=CMakeFiles\lz4cli.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100183~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100183~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1426~1.288\bin\Hostx64\x64\link.exe  CMakeFiles\lz4cli.dir\C_\lz4\programs\bench.c.obj CMakeFiles\lz4cli.dir\C_\lz4\programs\lz4cli.c.obj CMakeFiles\lz4cli.dir\C_\lz4\programs\lz4io.c.obj CMakeFiles\lz4cli.dir\C_\lz4\programs\datagen.c.obj  /out:lz4.exe /implib:lz4.lib /pdb:lz4.pdb /version:0.0  /machine:x64 /MACHINE:X64 /DEBUG /debug /INCREMENTAL /subsystem:console  lz4.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
 Visual Studio Incremental Link with embedded manifests
 Create CMakeFiles\lz4cli.dir/manifest.rc
 Create empty: CMakeFiles\lz4cli.dir/embed.manifest
 RC Pass 1:
 C:\PROGRA~2\WI3CF2~1\10\bin\100183~1.0\x64\rc.exe /fo CMakeFiles\lz4cli.dir/manifest.res CMakeFiles\lz4cli.dir/manifest.rc
 LINK Pass 1:
 C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1426~1.288\bin\Hostx64\x64\link.exe CMakeFiles\lz4cli.dir\C_\lz4\programs\bench.c.obj CMakeFiles\lz4cli.dir\C_\lz4\programs\lz4cli.c.obj CMakeFiles\lz4cli.dir\C_\lz4\programs\lz4io.c.obj CMakeFiles\lz4cli.dir\C_\lz4\programs\datagen.c.obj /out:lz4.exe /implib:lz4.lib /pdb:lz4.pdb /version:0.0 /machine:x64 /MACHINE:X64 /DEBUG /debug /INCREMENTAL /subsystem:console lz4.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:CMakeFiles\lz4cli.dir/intermediate.manifest CMakeFiles\lz4cli.dir/manifest.res
 LINK Pass 1: command "C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1426~1.288\bin\Hostx64\x64\link.exe CMakeFiles\lz4cli.dir\C_\lz4\programs\bench.c.obj CMakeFiles\lz4cli.dir\C_\lz4\programs\lz4cli.c.obj CMakeFiles\lz4cli.dir\C_\lz4\programs\lz4io.c.obj CMakeFiles\lz4cli.dir\C_\lz4\programs\datagen.c.obj /out:lz4.exe /implib:lz4.lib /pdb:lz4.pdb /version:0.0 /machine:x64 /MACHINE:X64 /DEBUG /debug /INCREMENTAL /subsystem:console lz4.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:CMakeFiles\lz4cli.dir/intermediate.manifest CMakeFiles\lz4cli.dir/manifest.res" failed (exit code 1104) with the following output:
 Microsoft (R) Incremental Linker Version 14.26.28805.0
 Copyright (C) Microsoft Corporation.  All rights reserved.
 LINK : lz4.exe not found or not built by the last incremental link; performing full link
 LINK : fatal error LNK1104: cannot open file 'lz4.lib'
 ninja: build stopped: subcommand failed.
```

This must be in some way dependent on building the command line tool `lz4.exe`, because when disabling all command line tools the build works.

--- a/build/cmake/CMakeLists.txt
+++ b/build/cmake/CMakeLists.txt
@@ -107,6 +107,10 @@ if(BUILD_SHARED_LIBS)
     OUTPUT_NAME lz4
     SOVERSION "${LZ4_VERSION_MAJOR}"
     VERSION "${LZ4_VERSION_STRING}")
+  if(MSVC)
+    target_compile_definitions(lz4_shared PRIVATE
+      LZ4_DLL_EXPORT=1)
+  endif()
   list(APPEND LZ4_LIBRARIES_BUILT lz4_shared)
 endif()
 if(BUILD_STATIC_LIBS)

